pkgs/tools/admin/nomachine-client/default.nix

```diff
@@ -49,6 +49,9 @@ in
       cp -r bin lib share $out/NX/
 
       ln -s $out/NX/bin $out/bin
+      cp ${lib.getLib systemd}/lib/libudev.so $out/NX/lib/libudev.so.0
+      cp ${lib.getLib boost}/lib/libboost_system.so $out/NX/lib/libboost_system.so.1.53.0
+      cp ${lib.getLib boost}/lib/libboost_thread.so $out/NX/lib/libboost_thread.so.1.53.0
```

Comment:
This is my attempt to address the following build error. Not sure if it's correct, but it builds and basic remote desktop functionality seems to work fine for me.

```
autoPatchelfHook could not satisfy dependency libboost_thread.so.1.53.0 wanted by /nix/store/7hvl1gpsw22l1q6nr8yk5df760s93ina-nomachine-client-7.0.211/NX/bin/nxusbd-legacy
autoPatchelfHook could not satisfy dependency libudev.so.0 wanted by /nix/store/7hvl1gpsw22l1q6nr8yk5df760s93ina-nomachine-client-7.0.211/NX/bin/nxusbd-legacy
autoPatchelfHook could not satisfy dependency libboost_system.so.1.53.0 wanted by /nix/store/7hvl1gpsw22l1q6nr8yk5df760s93ina-nomachine-client-7.0.211/NX/bin/nxusbd-legacy
Add the missing dependencies to the build inputs or set autoPatchelfIgnoreMissingDeps=true
builder for '/nix/store/xnkdnwp7nwy3z66i5lc61q1grb4pmjsx-nomachine-client-7.0.211.drv' failed with exit code 1
error: build of '/nix/store/xnkdnwp7nwy3z66i5lc61q1grb4pmjsx-nomachine-client-7.0.211.drv' failed
```

Reply:
Yeah looks like while it makes it build it doesn't actually make `nxusbd-legacy` work.

```
❯ result/bin/nxusbd-legacy                                                                                                                                                                       ~/Documents/nixpkgs
result/bin/nxusbd-legacy: /nix/store/i1ip8ywhr89ni1xidv4w1d3s9s0npgyd-nomachine-client-7.0.211/NX/lib/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /nix/store/i1ip8ywhr89ni1xidv4w1d3s9s0npgyd-nomachine-client-7.0.211/NX/lib/libboost_thread.so.1.53.0)
result/bin/nxusbd-legacy: /nix/store/i1ip8ywhr89ni1xidv4w1d3s9s0npgyd-nomachine-client-7.0.211/NX/lib/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /nix/store/i1ip8ywhr89ni1xidv4w1d3s9s0npgyd-nomachine-client-7.0.211/NX/lib/libboost_thread.so.1.53.0)
result/bin/nxusbd-legacy: /nix/store/i1ip8ywhr89ni1xidv4w1d3s9s0npgyd-nomachine-client-7.0.211/NX/lib/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /nix/store/i1ip8ywhr89ni1xidv4w1d3s9s0npgyd-nomachine-client-7.0.211/NX/lib/libboost_thread.so.1.53.0)
```

Reply:
autoPatchelfHook should detect it if you add boost to buildInputs.

Reply:
Sorry, should have mentioned I tried that, same error.

Reply:
Thats because we do not have ``libboost_thread.so.1.53.0`` in master. For libudev you need to use libudev0-shim.

Reply:
Does that 1.53 mean the package we'd need to have in master would be `boost153`? If so, yeah the lowest version in nixpkgs is `boost155`. And in that case, what's the suggested path forward? Add a `boost153` or just somehow kill the `nxusbd-legacy` support in this package?

Reply:
On Arch, with the same nomachine version installed I don't end up with a `nxusbd-legacy` binary installed. So I figure it's not essential.

Reply:
The `-legacy` version is not needed; according to the install scripts, it's for systems with linux kernel <3, so definitely not us. Just get rid of it and that should solve the issue :)

Reply:
Same goes for the `-legacy.so` file, btw.

Reply:
Thanks guys, I see that in the script now. PR updated.